scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -164,123 +195,217 @@
+  [compute-cluster k8s-actual-state-dict]
+  ; TODO: Should mark mea culpa retry
+  (kill-pod-in-weird-state compute-cluster (pod-has-just-completed compute-cluster k8s-actual-state-dict) k8s-actual-state-dict))
 
 (defn process
-  "Visit this pod-name, processing the new level-state. Returns the new expected state. Returns
+  "Visit this pod-name, processing the new level-state. Returns the new cook expected state. Returns
   empty dictionary to indicate that the result should be deleted. NOTE: Must be invoked with the lock."
-  [{:keys [api-client existing-state-map expected-state-map name] :as kcc} ^String pod-name]
-  (loop [{:keys [expected-state] :as expected-state-dict} (get @expected-state-map pod-name)
-         {:keys [synthesized-state pod] :as existing-state-dict} (get @existing-state-map pod-name)]
+  [{:keys [api-client k8s-actual-state-map cook-expected-state-map name] :as compute-cluster} ^String pod-name]
+  (loop [{:keys [cook-expected-state] :as cook-expected-state-dict} (get @cook-expected-state-map pod-name)
+         {:keys [synthesized-state pod] :as k8s-actual-state-dict} (get @k8s-actual-state-map pod-name)]
     (log/info "In compute cluster" name ", processing pod" pod-name ";"
-              "expected:" (prepare-expected-state-dict-for-logging expected-state-dict) ","
-              "existing:" (prepare-existing-state-dict-for-logging existing-state-dict))
-    ; TODO: We added an :expected/starting state to the machine, to represent when a pod is starting. We map instance.status/unknown to that state
-    ; The todo is to add in cases for [:expected/starting *] for those other states.
+              "cook-expected:" (prepare-cook-expected-state-dict-for-logging cook-expected-state-dict) ","
+              "k8s-actual:" (prepare-k8s-actual-state-dict-for-logging k8s-actual-state-dict))
+    ; We should have the cross product of
+    ;      :cook-expected-state/starting :cook-expected-state/running :cook-expected-state/completed :cook-expected-state/killed :missing
+    ; and
+    ;      :pod/waiting :pod/running :pod/succeeded :pod/failed :pod/unknown :missing
+    ;
+    ; for a total of 30 states.
+    ;
+    ; The only terminal expected states are :cook-expected-state/completed and :missing
+    ; The only terminal pod states are :pod/succeeded :pod/failed and :missing. We also treat :pod/unknown as a terminal state.
+
+    ; If you ignore the reloading on startup, the initial state is set at (:cook-expected-state/starting, :missing) when we first add a pod to launch.
+    ; The final state is (:missing, :missing)
+    ;
+    ; We use :cook-expected-state/killed to represent a user-chosen kill. If we're doing a state-machine-induced kill (because something
+    ; went wrong) it should occur by deleting the pod, so we go, e.g.,  (:running,:waiting) (an illegal state) to (:running,:missing)
+    ; to (:completed, :missing), to (:missing,missing) to deleted. We put a flag on when we delete so that we can indicate
```

Comment:
What does "put a flag on" mean?

Reply:
Indicates how we're going to solve some mea culpa retry and failure reason tracking.

Reply:
I still don't understand what "put a flag on" means, is it something we're doing in the code currently?

Reply:
No, not currently.

Reply:
Why do we say "We put a flag on when we delete" if it's not something we're doing currently?

Reply:
@scrosby Thoughts on this?

Reply:
@scrosby Was this addressed?